src/cli/commands/add.js

```diff
@@ -32,6 +36,10 @@ export class Add extends Install {
     ]
       .filter(Boolean)
       .shift();
+
+    if (flags.existing) {
+      this.flagToOrigin = '';
+    }
```

Comment:
Why is this required? Isn't it already handled by [this line](https://github.com/kaylieEB/yarn/blob/93b2aaeb33aad23218177f3a1ed2bd9870e4d077/src/cli/commands/add.js#L196) (according to the comment above, `depType` is the dependency source when upgrading a package)? I think it also might break the warning [here](https://github.com/kaylieEB/yarn/blob/93b2aaeb33aad23218177f3a1ed2bd9870e4d077/src/cli/commands/add.js#L132-L134), since the string will now always be empty

Reply:
So the issue I was running into was that when I ran `yarn upgrade typescript` (a dev dependency) in the workspace root directory, it would incorrectly warn me that workspaces prefer dev dependencies. This is because `flagsToOrgin` defaults to being a hard dependency when flags aren't passed. I don't think yarn should ever throw that warning if the command is an upgrade and save it for when it's a brand new add (hence the check for `flags.existing` passed in the upgrade file below).

Reply:
An inline comment about this would have been great.